File: internal/model/item.go
package model

import (
	"time"
)

// Item 物品模型
type Item struct {
	ID          uint      `gorm:"primaryKey"`
	UserID      uint      `gorm:"index;not null"`              // 所属用户 ID
	CategoryID  uint      `gorm:"index;not null"`             // 所属分类 ID
	Name        string    `gorm:"size:200;not null"`          // 物品名称
	Description string    `gorm:"size:500"`                   // 物品描述
	Quantity    int       `gorm:"default:1"`                 // 数量
	Unit        string    `gorm:"size:20"`                    // 单位
	ExpiredAt   time.Time `gorm:"index;not null"`             // 过期时间
	RemindDays  int       `gorm:"default:3"`                  // 提前提醒天数
	Status      int8      `gorm:"default:1;index"`           // 状态：1正常 2已过期 3已消耗
	CreatedAt   time.Time
	UpdatedAt   time.Time
}

// TableName 指定表名
func (Item) TableName() string {
	return "items"
}

File: src/model/item.ts
import { sqliteTable, text, integer, index } from "drizzle-orm/sqlite-core";

export const items = sqliteTable(
  "items",
  {
    id: integer("id").primaryKey({ autoIncrement: true }),
    userId: integer("user_id").notNull(),
    categoryId: integer("category_id").notNull(),
    name: text("name", { length: 200 }).notNull(),
    description: text("description", { length: 500 }),
    quantity: integer("quantity").notNull().default(1),
    unit: text("unit", { length: 20 }),
    expiredAt: text("expired_at").notNull(),
    remindDays: integer("remind_days").notNull().default(3),
    status: integer("status").notNull().default(1),
    createdAt: text("created_at").notNull(),
    updatedAt: text("updated_at").notNull(),
  },
  (table) => ({
    userIdIdx: index("idx_items_user_id").on(table.userId),
    categoryIdIdx: index("idx_items_category_id").on(table.categoryId),
    statusIdx: index("idx_items_status").on(table.status),
    expiredAtIdx: index("idx_items_expired_at").on(table.expiredAt),
    userStatusExpiredIdx: index("idx_items_user_status_expired").on(
      table.userId,
      table.status,
      table.expiredAt,
    ),
  }),
);

export type Item = typeof items.$inferSelect;
export type InsertItem = typeof items.$inferInsert;
